drivers/include/ata8510.h

```diff
@@ -0,0 +1,210 @@
+/*
+ * Copyright (C) 2020 Antonio Galea <antonio.galea@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser General
+ * Public License v2.1. See the file LICENSE in the top level directory for more
+ * details.
+ */
+
+/**
+ * @defgroup    drivers_ata8510 ATA8510 driver
+ *
+ * This module contains a driver for the radio device Atmel ATA8510.
+ *
+ * @{
+ *
+ * @file
+ * @brief       Interface definition for ATA8510 driver
+ *
+ * @author      Antonio Galea <antonio.galea@gmail.com>
+ * @author      Roberto Asquini <bobasquins@gmail.com>
+ */
+
+#ifndef ATA8510_H
+#define ATA8510_H
+
+#include <stdint.h>
+#include <inttypes.h>
+#include "xtimer.h"
+#include "periph/gpio.h"
+#include "periph/spi.h"
+
+#include "ata8510_internals.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief return codes
+ */
+enum {
+    ATA8510_OK = 0,                 /**< Everything is fine */
+    ATA8510_ERR_SPI,                /**< An error occurred with SPI */
+    ATA8510_ERR_MALLOC,             /**< An error occurred during malloc */
+    ATA8510_ERR_SIGNATURE,          /**< Wrong device signature found */
+    ATA8510_ERR_GPIO_INT,           /**< An error occurred with interrupt pin */
+    ATA8510_ERR_GPIO_RESET,         /**< An error occurred with reset pin */
+    ATA8510_ERR_GPIO_POWER,         /**< An error occurred with power pin */
+    ATA8510_ERR_EVENT_TIMEOUT,      /**< An expected event never happened */
+    ATA8510_ERR_BUFFER_OVERFLOW,    /**< Received too many bytes */
+    ATA8510_ERR_RECV_THREAD,        /**< Cannot create the receiver thread */
+};
+
+/**
+ * @brief struct holding all params needed for device initialization
+ */
+typedef struct ata8510_params {
+    spi_t spi;                              /**< SPI device */
+    spi_clk_t spi_clk;                      /**< SPI clock speed */
+    gpio_t cs_pin;                          /**< Chip select pin */
+    gpio_t int_pin;                         /**< IRQ pin */
+    gpio_t sleep_pin;                       /**< Sleep pin */
+    gpio_t reset_pin;                       /**< Reset pin */
+} ata8510_params_t;
+
+/**
+ * @brief   Packet handler callback
+ *
+ * NOTE: the callback is responsible for freeing buffer
+ *       after using it.
+ *
+ * @param[in] buffer    The packet received
+ * @param[in] len       Packet length
+ * @param[in] dbm       Signal strength in dBm
+ */
+typedef void (*ata8510_recv_cb_t)(uint8_t *buffer, int len, int dbm);
+
+/**
+ * @brief device descriptor
+ */
+typedef struct {
+    ata8510_params_t params;        /**< parameters for initialization */
+    uint8_t internal_state;         /**< current state of the radio */
+    mutex_t tx_lock;                /**< mutex for tx events */
+    mutex_t rx_lock;                /**< mutex for rx events */
+    ata8510_events_t *events;       /**< radio events registers */
```

Comment:
```suggestion
    ata8510_events_t events;       /**< radio events registers */
```

no need for `malloc`

Reply:
Done.